File: python/mission-to-mars-solid/src/mars/Helicoptere.py
from src.mars.Direction import Direction
from src.mars.ModuleMartien3D import ModuleMartien3D
from src.mars.Position import Position
from src.mars.Rover import Rover


class Helicoptere(Rover, ModuleMartien3D):

    def __init__(self, direction: Direction, position: Position):
        super().__init__(direction, position)

    def monter(self) -> None:
        self.position = Position(self.position.x, self.position.y, self.position.z + 1)

    def descendre(self) -> None:
        self.position = Position(self.position.x, self.position.y, self.position.z - 1)

File: java/mission-to-mars-solid/src/main/java/mars/Helicoptere.java
package mars;

public class Helicoptere extends Rover implements ModuleMartien3D {

	public Helicoptere(Direction direction, Position position) {
		super(direction, position);
	}

	@Override
	public void monter() {
		position = new Position(position.getX(), position.getX(), position.getZ() + 1);
	}

	@Override
	public void descendre() {
		position = new Position(position.getX(), position.getX(), position.getZ() - 1);
	}

}